lib/rom/sql/function.rb

```diff
@@ -132,6 +132,26 @@ def cast(expr, db_type = TypeSerializer[:default].call(type))
         Attribute[type].meta(sql_expr: ::Sequel.cast(expr, db_type))
       end
 
+      # Add a CASE clause for handling if/then logic
+      #
+      # @example
+      #   users.select { string::case(status, { "active" => true }, false).as(:activated) }
+      #   users.select { string::case({ "active" => true }, false).as(:activated) }
+      #   users.select { string::case([[{ status: ["active", nil]}, true]], false) }
+      #
+      # @param [ROM::SQL::Attribute] expr Expression consist of 3 args - conditions, default value and optional expression
+      #
+      # @return [ROM::SQL::Attribute]
+      #
+      # @api public
+      def case(*args)
+        if args.first.is_a?(Hash) || args.first.is_a?(Array)
+          Attribute[type].meta(sql_expr: ::Sequel.case(args[0], args[1]))
+        else
+          Attribute[type].meta(sql_expr: ::Sequel.case(args[1], args[2], args[0]))
+        end
+      end
```

Comment:
Seems like changing this method's sig to `def case(*args, **opts)` should simplify things, no?

Reply:
Thanks for review @solnic !
Very good point, I wanted to omit `if else` statement, but I cannot achieve that because of sequel implementation of CaseExpression are pretty not friendly for wrappers ;)

This looking good for you?
```
      def case(*args, **opts)
        if opts[:expr]
          Attribute[type].meta(sql_expr: ::Sequel.case(*args, opts[:expr]))
        else
          Attribute[type].meta(sql_expr: ::Sequel.case(*args))
        end
      end
```

Reply:
Yeah, that's what I meant :) I knew we can't avoid the conditional.